backend/app/api/endpoints/admin/public_bots.py

```
@@ -154,9 +160,84 @@ def _validate_bot_resource_references(
+                is_custom = model_spec.get("isCustomConfig", False)
+                if is_custom:
+                    # Custom model - return modelConfig with protocol
+                    model_config = model_spec.get("modelConfig", {})
+                    protocol = model_spec.get("protocol")
+                    if protocol:
+                        agent_config = {**model_config, "protocol": protocol}
+                    else:
+                        agent_config = model_config
+                else:
+                    # Predefined model - return bind_model reference
+                    metadata = model.json.get("metadata", {})
```

Ruff (0.14.14)

Local variable `metadata` is assigned to but never used

Remove assignment to unused variable `metadata`

(F841)